circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/Lifecycle.kt

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.foundation
+
+import androidx.compose.runtime.ProvidableCompositionLocal
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.staticCompositionLocalOf
+
+public interface Lifecycle {
```

Comment:
I've put zero effort into this API so far. I was thinking of adding a `LifecycleState` enum (replacing the `isPaused: Boolean`). Open to suggestions though.

Reply:
I think starting with a boolean for now is fine. Should we annotate it as `@Stable` too?